### ISSUE TITLE
Assigning & returns a parsing error

### ISSUE THREAD

**Issue by kaelig** [sasstools/sass-lint#385]
Using sass-lint cli 1.3.2.

Both these declarations are setting the parser off:

``` scss
$parent: #{&};
$parent: &;
```

Stacktrace:

```
/Users/kdeloumeauprigent/workspace/salesforce/design-system-internal/node_modules/sass-lint/lib/groot.js:21
    throw new Error('Parsing error' + fileInfo + ': ' + e.message);
    ^

Error: Parsing error at ui/dependencies/core.scss: Please check validity of the block starting from line #10
    at module.exports (/Users/kdeloumeauprigent/workspace/salesforce/design-system-internal/node_modules/sass-lint/lib/groot.js:21:11)
    at Function.sassLint.lintText (/Users/kdeloumeauprigent/workspace/salesforce/design-system-internal/node_modules/sass-lint/index.js:36:13)
    at /Users/kdeloumeauprigent/workspace/salesforce/design-system-internal/node_modules/sass-lint/index.js:91:21
    at Array.forEach (native)
    at Function.sassLint.lintFiles (/Users/kdeloumeauprigent/workspace/salesforce/design-system-internal/node_modules/sass-lint/index.js:90:9)
    at detectPattern (/Users/kdeloumeauprigent/workspace/salesforce/design-system-internal/node_modules/sass-lint/bin/sass-lint.js:15:18)
    at Object.<anonymous> (/Users/kdeloumeauprigent/workspace/salesforce/design-system-internal/node_modules/sass-lint/bin/sass-lint.js:77:3)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
```

**Comment by DanPurdy** [comment]
I'll look into whether this is still the case with the current build of `gonzales-pe`. 

Thanks for the report!

**Comment by bgriffith** [comment]
Could you drop an example of this in use? Thanks!

**Comment by kaelig** [comment]
Ok, this is hypothetical code but still close to my real world use case:

`ˋˋscss

@mixin parent() {
  @if (&) {
    $parent: &;
    @warn '#{$parent}';
  }
}

.foo {
  @include parent { bar: baz; }
}
`ˋ`

**Comment by DanPurdy** [comment]
So the original issue is fixed, but the example is still failing with the AST 😢

**Comment by bgriffith** [comment]
This is fixed in the newly released version v1.11.0. Please let me know if you still have issues. Thanks!
